homeassistant/components/comelit/switch.py

```diff
@@ -24,21 +33,92 @@ async def async_setup_entry(
 ) -> None:
     """Set up Comelit switches."""
 
+    if config_entry.data.get(CONF_TYPE, BRIDGE) == BRIDGE:
+        await async_setup_bridge_entry(hass, config_entry, async_add_entities)
+    else:
+        await async_setup_vedo_entry(hass, config_entry, async_add_entities)
+
+
+async def async_setup_bridge_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Comelit Bridge switches."""
+
     coordinator: ComelitSerialBridge = hass.data[DOMAIN][config_entry.entry_id]
 
-    entities: list[ComelitSwitchEntity] = []
+    entities: list[ComelitBridgeSwitchEntity] = []
     entities.extend(
-        ComelitSwitchEntity(coordinator, device, config_entry.entry_id)
+        ComelitBridgeSwitchEntity(coordinator, device, config_entry.entry_id)
         for device in coordinator.data[IRRIGATION].values()
     )
     entities.extend(
-        ComelitSwitchEntity(coordinator, device, config_entry.entry_id)
+        ComelitBridgeSwitchEntity(coordinator, device, config_entry.entry_id)
         for device in coordinator.data[OTHER].values()
     )
     async_add_entities(entities)
 
 
-class ComelitSwitchEntity(CoordinatorEntity[ComelitSerialBridge], SwitchEntity):
+async def async_setup_vedo_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Comelit vedo switches."""
+
+    coordinator: ComelitVedoSystem = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        ComelitVedoSwitchEntity(coordinator, device, config_entry.entry_id)
+        for device in coordinator.data[ALARM_ZONES].values()
+    )
+
+
+class ComelitVedoSwitchEntity(CoordinatorEntity[ComelitVedoSystem], SwitchEntity):
+    """Switch device."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self,
+        coordinator: ComelitVedoSystem,
+        device: ComelitVedoZoneObject,
+        config_entry_entry_id: str,
+    ) -> None:
+        """Init switch entity."""
+        self._api = coordinator.api
+        self._device = device
+        super().__init__(coordinator)
+        # Use config_entry.entry_id as base for unique_id
+        # because no serial number or mac is available
+        self._attr_unique_id = f"{config_entry_entry_id}-switch-{device.index}"
+        self._attr_device_info = coordinator.platform_device_info(device, "zone")
+        self._attr_device_class = SwitchDeviceClass.SWITCH
+
+    async def _switch_set_state(self, state: int) -> None:
+        """Set desired switch state."""
+        await self.coordinator.api.set_zone_status(
+            self._device.index + 1, "incl" if state == STATE_ON else "excl"
```

Comment:
Why +1 ?

Reply:
also in js from comelit web interface there is +1 on the index, i've verified and this +1 is necessary to match the correct device

```
if ((status & 128) == 0) {
	buttons[lang75] = function() { //Escludi
		sendVedoCommand('excl', parseInt(zona)+1);
		$("#zone_modify").dialog("close");
	};
} else {
	buttons[lang76] = function() { // Includi
		sendVedoCommand('incl', parseInt(zona)+1);
		$("#zone_modify").dialog("close");
	};
}
```